common/configuration/1password_provider.go

```diff
@@ -27,172 +25,156 @@ func (OnePasswordProvider) Role() Secrets                               { return
-		return nil, fmt.Errorf("error decoding op single response: %w", err)
-	}
-	return &single, nil
+func (i item) value(field string) (string, bool) {
+	secret, ok := slices.Find(i.Fields, func(item entry) bool {
+		return item.ID == field
+	})
+	return secret.Value, ok
 }
 
-// Custom parser for 1Password secret references because the format is not a standard URL, and we also need to
-// allow users to omit the field name so that we can support secrets with multiple fields.
-//
-// Does not support "section-name".
-//
-//	op://<vault-name>/<item-name>[/<field-name>]
-//
-// Secret references are case-insensitive and support the following characters:
-//
-//	alphanumeric characters (a-z, A-Z, 0-9), -, _, . and the whitespace character
-//
-// If an item or field name includes a / or an unsupported character, use the item
-// or field's unique identifier (ID) instead of its name.
-//
-// See https://developer.1password.com/docs/cli/secrets-reference-syntax/
-type secretRef struct {
-	Vault string
-	Item  string
-	Field optional.Option[string]
-}
+// op --format json item get --vault Personal "With Spaces"
+func getItem(ctx context.Context, vault string, ref Ref) (*item, error) {
+	logger := log.FromContext(ctx)
 
-var validCharsRegex = regexp.MustCompile(`^[a-zA-Z0-9\-_\. ]+$`)
+	args := []string{"--format", "json", "item", "get", "--vault", vault, ref.String()}
+	output, err := exec.Capture(ctx, ".", "op", args...)
+	logger.Debugf("Getting item with args %v", args)
+	if err != nil {
+		if strings.Contains(string(output), "isn't a vault") {
+			return nil, fmt.Errorf("vault %q not found: %w", vault, err)
+		}
 
-func decodeSecretRef(ref string) (*secretRef, error) {
+		// Item not found, seen two ways of reporting this:
+		if strings.Contains(string(output), "not found in vault") {
+			return nil, notFoundError{vault, ref}
+		}
+		if strings.Contains(string(output), "isn't an item") {
+			return nil, notFoundError{vault, ref}
+		}
 
-	// Take out and check the "op://" prefix
-	const prefix = "op://"
-	if !strings.HasPrefix(ref, prefix) {
-		return nil, fmt.Errorf("must start with \"op://\"")
+		return nil, fmt.Errorf("run `op` with args %v: %w", args, err)
 	}
-	ref = ref[len(prefix):]
-
-	parts := strings.Split(ref, "/")
 
-	if len(parts) < 2 {
-		return nil, fmt.Errorf("must have at least 2 parts")
-	}
-	if len(parts) > 3 {
-		return nil, fmt.Errorf("must have at most 3 parts")
+	var full item
+	if err := json.Unmarshal(output, &full); err != nil {
+		return nil, fmt.Errorf("error decoding op full response: %w", err)
 	}
+	return &full, nil
+}
 
-	for _, part := range parts {
-		if part == "" {
-			return nil, fmt.Errorf("url parts must not be empty")
-		}
-
-		if !validCharsRegex.MatchString(part) {
-			return nil, fmt.Errorf("url part %q contains unsupported characters. regex: %q", part, validCharsRegex)
-		}
+// op item create --category Password --vault FTL --title mod.ule "password=val ue"
```

Comment:
Does this correctly deal with the raw JSON values we send, or do we need to b64 encode them first?

Reply:
I did try lots of different chars with no problem, even an emoji worked.